### ISSUE TITLE
CTRL+C does not interrupt batch script with PAUSE command

### ISSUE THREAD

**Issue by MX9000** [joncampbell123/dosbox-x#4479]
### Describe the bug

Trying to install the game Rick Dangerous (https://archive.org/details/003032-RickDangerous) I noticed that the INSTALL.BAT batch script does not stop when pressing the CTRL+C key combination and the files are still copied to hard disk.

![rick_dangerous_setup](https://github.com/joncampbell123/dosbox-x/assets/58113425/c8e333cc-8449-4995-add9-4fb4d376733a)

Under PCem with MS-DOS 5 and 6 after pressing CTRL+C the question "Terminate batch job (Y/N)?" appears and pressing the Y key stops the batch procedure.

### Steps to reproduce the behaviour

1. Launch INSTALL.BAT
2. Try to stop the installation with CTRL+C key combination when offered by the script

### Expected behavior

In real MS-DOS 5 & 6 the batch script stops.

### What operating system(s) this bug have occurred on?

Windows 10 Pro 22H2

### What version(s) of DOSBox-X have this bug?

DOSBox-X 2023.09.01 VSbuild & MinGW 32 bit

### Used configuration

```ini
Standard
```

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by Laci60** [comment]
Hi!
I have installed dosbox-x-windows-20231001165402-setup.exe and Ctrl+c does not work despite of commit 1168cfb.
Op: Win10 22H2
Batch file:
    CLS
    BREAK ON
    ECHO Ctrl+C or not Ctrl+C
    PAUSE
    It isn't Ctrl+C
    PAUSE

Result:
    Ctrl+C or not Ctrl+C
    Press any key to continue . . .
    ^C
    t isn't Ctrl+C
    Press any key to continue . . .
    C:\

SDL1, SDL2, MinGW, VBS versions all produce this.

**Comment by joncampbell123** [comment]
> Hi!
> 
> I have installed dosbox-x-windows-20231001165402-setup.exe and Ctrl+c does not work despite of commit [1168cfb](https://github.com/joncampbell123/dosbox-x/commit/1168cfb3ab21c2c7d16aa118a41a75cf961b7096). Op: Win10 22H2 Hungarian, keyb HU,852 Dosbox config: standard with keyb HU
> 
> Batch file: CLS BREAK ON ECHO Ctrl+C or not Ctrl+C PAUSE It isn't Ctrl+C PAUSE
> 
> Result: Ctrl+C or not Ctrl+C Press any key to continue . . . ^C t isn't Ctrl+C Press any key to continue . . . C:\
> 
> Yes, "t" instead of "It"...
> 
> SDL1, SDL2, MinGW, VBS versions all produce this.

Yes. I got some of the work done but I'm still having trouble getting CTRL+C to interrupt pause. Primarily the DOS console emulation only checks for CTRL+C when reading from STDIN, when it's supposed to check also on console output. That part isn't done yet.

**Comment by joncampbell123** [comment]
Actually to put it another way, CTRL+C checking is done within INT 21h for input and output, but the PAUSE command is built-in and writes to the console without INT 21h which is why CTRL+C doesn't work right.
